packages/web/hooks/one-click-trading/use-one-click-trading-params.ts

```diff
@@ -39,90 +45,143 @@ export function getParametersFromOneClickTradingInfo({
-    setInitialTransaction1CTParams(nextTransaction1CTParams);
+    const hasChanges = changes.length > 0;
+    const wouldChange =
+      compareParams({
+        prev: draft,
+        next: sessionOrDefaultWithEnabled,
+      }).length > 0;
+
+    if (!hasChanges && wouldChange) {
+      reset();
+    }
   }, [
-    defaultIsOneClickEnabled,
-    defaultTransaction1CTParams,
-    oneClickTradingInfo,
-    transaction1CTParams,
+    sessionOrDefault,
+    draft,
+    sessionOrDefaultWithEnabled,
+    changes.length,
+    reset,
+    current,
   ]);
 
-  const reset = useCallback(() => {
-    const paramsToSet = oneClickTradingInfo
-      ? getParametersFromOneClickTradingInfo({
-          oneClickTradingInfo,
-          defaultIsOneClickEnabled,
+  const setTransaction1CTParamsWithChanges = useCallback(
+    (
+      newDraftOrFn: SetStateAction<OneClickTradingTransactionParams | undefined>
+    ) => {
+      if (!current) reset();
+
+      const nextDraft = runIfFn(newDraftOrFn, draft);
+
+      setDraft(nextDraft);
+
+      setChanges(() =>
+        compareParams({
+          prev: current,
+          next: nextDraft,
         })
-      : defaultTransaction1CTParams;
-    if (!paramsToSet && !initialTransaction1CTParams) return;
-
-    const nextTransaction1CTParams = paramsToSet
-      ? {
-          isOneClickEnabled: defaultIsOneClickEnabled,
-          ...paramsToSet,
-        }
-      : initialTransaction1CTParams;
-    setTransaction1CTParams(nextTransaction1CTParams);
-  }, [
-    defaultIsOneClickEnabled,
-    defaultTransaction1CTParams,
-    initialTransaction1CTParams,
-    oneClickTradingInfo,
-  ]);
+      );
+    },
+    [current, reset, draft]
+  );
 
   return {
-    transaction1CTParams,
-    setTransaction1CTParams,
-    spendLimitTokenDecimals:
-      defaultTransaction1CTParams?.spendLimitTokenDecimals,
+    changes,
+    initialTransaction1CTParams: current,
+    transaction1CTParams: draft,
+    spendLimitTokenDecimals: defaultParams?.spendLimitTokenDecimals,
     isLoading,
-    isError,
     reset,
+    setChanges,
+    setTransaction1CTParams: setTransaction1CTParamsWithChanges,
   };
-};
+}
+
+export type OneClickTradingParamsChanges = Array<
+  "spendLimit" | "sessionPeriod" | "isEnabled" | "networkFeeLimit"
+>;
+
+function compareParams({
+  prev,
+  next,
+}: {
+  prev?: OneClickTradingTransactionParams;
+  next?: OneClickTradingTransactionParams;
+}): OneClickTradingParamsChanges {
+  let changes: OneClickTradingParamsChanges = [];
+
+  if (prev?.spendLimit.toString() !== next?.spendLimit.toString()) {
+    changes.push("spendLimit");
+  }
+
+  if (prev?.sessionPeriod.end !== next?.sessionPeriod.end) {
+    changes.push("sessionPeriod");
+  }
+
+  if (prev?.isOneClickEnabled !== next?.isOneClickEnabled) {
+    changes.push("isEnabled");
+  }
+
+  if (prev?.networkFeeLimit !== next?.networkFeeLimit) {
+    changes.push("networkFeeLimit");
+  }
+
+  return changes;
+}
```

Comment:
I believe a similar method already exists in one-click-trading-settings. I recommend reusing it